salt/metalk8s/orchestrate/upgrade/etcd.sls

```diff
@@ -0,0 +1,52 @@
+{%- set dest_version = pillar.orchestrate.dest_version %}
+{%- set etcd_nodes = salt.metalk8s.minions_by_role('etcd') %}
+{%- set control_plane_ip = salt.saltutil.runner(
+    'mine.get',
+    tgt=pillar.bootstrap_id,
+    fun='control_plane_ip'
+)[pillar.bootstrap_id] %}
+
+{%- set pillar_data = {
+    'repo': {
+        'host': control_plane_ip
+    },
+    'registry_ip': control_plane_ip
+} %}
+
+Check etcd cluster health:
+  module.run:
+    - metalk8s.check_etcd_health:
+      - hostname: {{ pillar.bootstrap_id }}
+    - failhard: True
```

Comment:
I honestly don't like this. Would rather prefer to have a clear `require`-chain.

Reply:
Hmm it's harder to do a `require`-chain and also as a real point of view upgrade of one node not really require an upgrade of the previous node.

Also failhard made the output really clear here as if we have tones of nodes if we fail on the first one we will have a huge output with "One or more prequiriste states failed blabla` and only one with an error.

I can try to do a `require` chain but ... 

Edit: Or yes I agree for this one which is out of the for I should use a require I agree but for the last one at the end of the for I think it's better to keep the failhard